File: src/dotnet/Ansight.Tools.VisualTree/ClearOverlaysTool.cs
namespace Ansight.Tools.VisualTree;

public sealed class ClearOverlaysTool : ITool
{
    public string Category => "ui";

    public ToolPolicy Policy => ToolPolicy.Write;

    public string Id => VisualTreeToolIds.ClearOverlays;

    public string Name => "Clear Overlays";

    public string Description => "Removes all diagnostic overlays from the active app window.";

    public string Keywords => "ui overlay highlight clear remove all";

    public ToolSchema ArgumentsSchema => VisualTreeToolSchemas.ClearOverlaysArguments;

    public ToolSchema ResultSchema => VisualTreeToolSchemas.ClearOverlaysResult;

    public Task<ToolResult> Execute(IReadOnlyDictionary<string, string> arguments)
    {
        ArgumentNullException.ThrowIfNull(arguments);
        return VisualTreeSupport.ClearOverlaysAsync(arguments);
    }
}

File: src/ios/Sources/AnsightToolsVisualTree/ClearOverlaysTool.swift
import AnsightCore
import Foundation

public final class ClearOverlaysTool: AnsightTool {
    public init() {}

    public var descriptor: AnsightToolDescriptor {
        AnsightToolDescriptor(
            id: AnsightVisualTreeToolIds.clearOverlays,
            name: "Clear Overlays",
            description: "Removes all diagnostic overlays from the active app window.",
            category: "ui",
            policy: .write,
            keywords: "ui overlay highlight clear remove all",
            argumentsSchema: AnsightVisualTreeToolSchemas.clearOverlaysArguments,
            resultSchema: AnsightVisualTreeToolSchemas.clearOverlaysResult
        )
    }

    public func execute(arguments: [String: String]) throws -> AnsightToolExecutionResult {
        AnsightVisualTreeOverlaySupport.clearOverlays(arguments: arguments)
    }
}
